scripts/dataflow_copy.py

```diff
@@ -3,18 +3,76 @@
 
 import psycopg2
 import os
+import shutil
+import json
 
 from neurobooth_terra import Table, create_table
 from neurobooth_terra.fixes import OptionalSSHTunnelForwarder
 from neurobooth_terra.dataflow import copy_files
 
 from config import ssh_args, db_args
 
+
+# A note about setting the value of reserve_threshold in
+# the config json:
+# 
+# The reserve_threshold value must be in units of bytes
+# A specific TB value can be expressed as TB*1024**4
+# For example a reserve_threshold of 2TB can be expressed
+# as 2*1024**4 = 2199023255552
+# 
+# The value of reserve_threshold should be chosen such that
+# it is 3 to 4 times the amount of data generated per session,
+# and leaves about 2% of the total disk memory empty
+# 
+# A note about suitable_volumes in the config json:
+# The destination directory paths should exist.
+
+
+def get_volume_to_fill(volumes: list, threshold: int) -> str:
+    ''' Return the most filled volume that is below reserve_threshold
+    '''
+    vol_disk_usage = {}
+    for vol in volumes:
+        stats = shutil.disk_usage(vol)
+        if stats.free > threshold:
+            vol_disk_usage[vol] = stats.used
+
+    if len(vol_disk_usage) <= 1:
+        raise ValueError(f'Only {len(vol_disk_usage)} available volume left, aborting copy till more volumes are added')
+
+    return max(vol_disk_usage, key=vol_disk_usage.get)
+
+
+def check_if_copied(session: str, volumes: list) -> tuple:
+    ''' Checks if a session is already copied to one of the volumes,
+        and returns the volume path if copied
+    '''
+    for vol in volumes:
+        if os.path.exists(os.path.join(vol, session)):
+            return True, os.path.join(vol, session)
+    return False, ''
+
+
+configs = json.load(open('../dataflow_config.json'))
+suitable_volumes = configs['suitable_volumes']  # list
+reserve_threshold = configs['reserve_threshold']  # int
+
+
+# ---- Printing disk usage statistics ---- #
+print("Disk Usage Statistics\n")
+print("Volume\t\t\t\t\tTotal (TB)\tUsed (TB)\tAvailable (TB)")
+for vol in suitable_volumes:
+    stats = shutil.disk_usage(vol)
+    print(vol, round(stats.total/1024**4, 2), round(stats.used/1024**4, 2), round(stats.free/1024**4, 2), sep='\t\t')
```

Comment:
convert the 1024**4 to a (global?) constant called TB or TERRABYTE.

Reply:
I would personally go with TERRABYTE = 1 << 40 (or 2**40) instead of scaling 1024

Reply:
Made these changes. Converted to 1024**4 with an inline comment # 1TB = 1024 bytes ** 4, as I feel that is more intuitive.